### PR TITLE
[MM-57826] Make sure the original errors are wrapped when AppErrors are returned

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Wrapping the error allows the upper layers to check for the wrapped error and allows for prettier printing,

The regex I used is `model.NewAppError\((.+), ([a-z,A-Z]+)\.Error\(\), (.+)\)` and the replacement is `model.NewAppError($1, "", $3).Wrap($2)`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57826

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/license.go [hunk 1]
```diff
@@ -267,7 +267,7 @@ func requestRenewalLink(c *Context, w http.ResponseWriter, r *http.Request) {
 	// check if it is possible to renew license on the portal with generated token
 	status, e := c.App.Cloud().GetLicenseSelfServeStatus(c.AppContext.Session().UserId, token)
 	if e != nil {
```
Comment (isacikgoz) [inline server/channels/api4/license.go:269]: First time seeing a single letter error var name :)

File: server/channels/api4/system.go [hunk 2]
```diff
@@ -811,10 +811,10 @@ func upgradeToEnterpriseStatus(c *Context, w http.ResponseWriter, r *http.Reques
 		var isErr *upgrader.InvalidSignature
 		switch {
 		case errors.As(err, &isErr):
-			appErr := model.NewAppError("upgradeToEnterpriseStatus", "api.upgrade_to_enterprise_status.app_error", nil, err.Error(), http.StatusBadRequest)
+			appErr := model.NewAppError("upgradeToEnterpriseStatus", "api.upgrade_to_enterprise_status.app_error", nil, "", http.StatusBadRequest).Wrap(err)
```
Comment (isacikgoz) [inline server/channels/api4/system.go:814]:
```suggestion
			appErr := model.NewAppError("upgradeToEnterpriseStatus", "api.upgrade_to_enterprise_status.app_error", nil, "", http.StatusBadRequest).Wrap(isErr)
```

**Review by isacikgoz — COMMENTED** [review]
LGTM, only one small thing.

File: server/channels/api4/system.go (hunk 2 above)
Comment (hanzei) [reply server/channels/api4/system.go:814]: Good catch :+1:

File: server/channels/api4/license.go (hunk 1 above)
Comment (hanzei) [reply server/channels/api4/license.go:269]: Error names in the Mattermost code base are special :star:

**Review by isacikgoz — COMMENTED** [review]
Awesome 👍

**Comment by hanzei** [comment]
@isacikgoz Did you meant to approve the PR?

**Review by isacikgoz — APPROVED** [review]
> @isacikgoz Did you mean to approve the PR?

Indeed, probably haven't changed the review status 🤦

**Review by agarciamontoro — APPROVED** [review]
Great, man, thanks!

**Pull request merged**